tsconfig.json

```diff
@@ -41,12 +41,10 @@
     "next-env.d.ts",
     "additional.d.ts",
     "src/**/*",
-    "**/*.ts",
-    "**/*.tsx",
-    "**/*.js",
-    "**/*.jsx"
+    "*/*.{ts|tsx|js|jsx}",
```

Comment:
the `**/*` globs here and in .eslintrc.json were causing the OOM for me. restricting them slightly, eslint travels through the files without hanging now.